tests/mainnet/metaregistry/integrative/test_getters.py

```diff
@@ -0,0 +1,793 @@
+    except brownie.exceptions.VirtualMachineError:
+        registry_query_reverts = True
+
+    if sum(registry.get_balances(pool)) == 0:
+        pytest.skip(f"Empty pool: {pool}")
+
+    if registry_query_reverts:
+        with brownie.reverts():
+            metaregistry.get_underlying_balances(pool)
+    else:
+        metaregistry_output = metaregistry.get_underlying_balances(pool)
+
+        if metaregistry.is_meta(pool):
+            assert metaregistry_output[2] > 0  # it must have a third coin
+        else:
+            assert metaregistry_output[1] > 0  # it must have a second coin
+
+        for idx, registry_value in enumerate(actual_output):
+            assert registry_value == metaregistry_output[idx]
+
+
+@pytest.mark.parametrize("pool_id", range(MAX_POOLS))
+def test_get_n_coins(metaregistry, registry_pool_index_iterator, pool_id):
+
+    skip_if_pool_id_gt_max_pools_in_registry(pool_id, registry_pool_index_iterator)
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    metaregistry_output = metaregistry.get_n_coins(pool)
+
+    # crypto factory does not have get_n_coins method
+    if registry_id != METAREGISTRY_CRYPTO_FACTORY_HANDLER_INDEX:
+
+        actual_output = registry.get_n_coins(pool)
+
+        # registry returns tuple, we want the first one (since the second)
+        # index is about basepool n coins
+        if not type(actual_output) == brownie.convert.datatypes.Wei:
+
+            actual_output = actual_output[0]
+
+        # registry returns 0 value for n coins: something's not right on the
+        # registry's side. find n_coins via registry.get_coins:
+        elif actual_output == 0:
+
+            coins = registry.get_coins(pool)
+            actual_output = sum([1 for coin in coins if coin != brownie.ZERO_ADDRESS])
+
+    else:
+
+        # do get_coins for crypto factory:
+        coins = registry.get_coins(pool)
+        actual_output = sum([1 for coin in coins if coin != brownie.ZERO_ADDRESS])
+
+    assert actual_output == metaregistry_output
+
+
+@pytest.mark.parametrize("pool_id", range(MAX_POOLS))
+def test_get_n_underlying_coins(metaregistry, registry_pool_index_iterator, pool_id):
+
+    skip_if_pool_id_gt_max_pools_in_registry(pool_id, registry_pool_index_iterator)
+
+    registry_id, registry_handler, registry, pool = registry_pool_index_iterator[pool_id]
+
+    metaregistry_output = metaregistry.get_n_underlying_coins(pool)
+
+    # check registry for n_coins first:
+    if registry_id in [
+        METAREGISTRY_STABLE_FACTORY_HANDLER_INDEX,
+        METAREGISTRY_STABLE_REGISTRY_HANDLER_INDEX,
+        METAREGISTRY_CRYPTO_REGISTRY_HANDLER_INDEX,
+    ]:
+
+        n_coins = registry.get_n_coins(pool)
+
+        if type(n_coins) == brownie.convert.datatypes.Wei:
+
+            try:
+                assert n_coins == metaregistry_output
+            except AssertionError:
```

Comment:
don't use exception handling so liberally in your tests. here you're handling the exception but only adding an assertion if the condition `coins[1] == BTC_BASEPOOL_LP_TOKEN_MAINNET` is met. That means that if `n_coins != metaregistry_output` on any other pool, no exception will be raised and your test will pass (when you clearly don't want it to).

instead just do a simple condition. you can hardcode the addresses of the pools instead of the lp token if `get_coins` can revert (or maybe call the metaregistry method instead).